.github/workflows/doc.yml

```diff
@@ -0,0 +1,68 @@
+name: Deploy Sphinx Doc to Pages
```

Comment:
Do we need a second workflow for this? 
Conceptually it feels like build and deploy are related steps, and "proximty-wise" there is less magic if they are in the file as well

Reply:
Think we might need to split them out if we want docs to be able to make on PR. pages.yml won't seem to run on a PR even if it's not going to deploy.